### ISSUE TITLE
An always on top windowrule for floating windows

### ISSUE THREAD

**Issue by Abaan404** [hyprwm/Hyprland#2382]
Hi, been using Hyprland for just over a week and have been enjoying it a lot so thanks for all your work!

I use the fullscreen mode of hyprland quite often and use apps (i.e. firefox's picture-in-picture mode or custom drop-down terminal from pyprland) that I would want to stay always on top of other windows, but they vanish when I enter fullscreen mode or render behind other floating windows if I switch focus (pin doesn't work if the application was already launched before).

I understand the idea of fullscreen is meant to keep only the said window front and center but in a practical scenario, it would be nice to be able to add a rule to keep certain windows on top.

tl;dr A windowrule to set floating windows to always render on top of other applications in fullscreen or floating mode.

**Comment by vaxerski** [comment]
> pin doesn't work if the application was already launched before

?

`hyprctl dispatch pin ...`?

**Comment by Abaan404** [comment]
I recorded a clip of me using it, actually looks like the window is there (hyprctl recognizes it as active) but it doesn't show itself on fullscreen until I click on it which I didn't realize before.
`hyprctl dispatch pin ...` doesn't seem to have any effect on this behavior other than toggling its pinned state

https://streamable.com/acygkd

**Comment by Abaan404** [comment]
Not sure when this happened but looks after I updated everything yesterday, seems that pinned windows are now properly being overlayed over other fullscreen and floating windows. Marking this as closed since this seems to be now implemented/fixed.
